### PR TITLE
github: Add a workflow to build cilium-cli binaries

### PR DESCRIPTION
**Pull request by michi-covalent**
Run the workflow on pull requests to ensure that cilium-cli builds on all the platforms.

### CODE REVIEW

File: .github/workflows/cilium-cli.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  cancel-in-progress: true
+
+jobs:
+  build-cilium-cli-binaries:
+    name: Build Cilium CLI binaries
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      - name: Setup go
```
Comment (sayboras) [inline .github/workflows/cilium-cli.yaml:22]: might need to add disk-cleanup step like other build workflow.

**Review by viktor-kurchenko — APPROVED** [review]

**Review by bimmlerd — APPROVED** [review]
CODEOWNERS lgtm

**Review by sayboras — APPROVED** [review]

**Pull request merged**